Converted status field to Submit/Save buttons

ashley opened the pull request:
For Issue #166 
* Removed status field from add and edit view. They would only appear if the status field was not found, which also means the view will use the original submit/save buttons.
* List view should still include status field; they should be updated as well
* Editing Draft events: Options to announce, and save changes will appear
* Editing Announced events: Options to cancel the event will appear
* Editing Canceled events: Options to un-cancel and delete events will appear
* Did basic QA testing of adding/editing events with different status.

Small bug: When updating add and edited resources, users have to refresh the list view to see the updates.

ashley committed:
Added edit page

ashley committed:
Added event-edit to page provider

ashley committed:
Added buttons for event-add

ashley committed:
Logic for event-add button

ashley committed:
Change status visibility to html

ashley committed:
Delete test function

ashley committed:
Debugging edit bug

ashley committed:
Debugging

ashley committed:
Fixed event edit controller

ashley committed:
Fixed event add ctrl

ashley committed:
Fixed annouced button

ashleytqy reviewed (commented):
this looks great!! 💯

a few design concerns –
when we are editing a `draft` event, having `save changes`, `announce event` and `delete event` as call to actions may be confusing. also, if we want to change the time & announce the event, we have to change the time, click `save changes`, go back to the edit page, before we can `announce event`.

a possible solution: on the edit page – instead of  `save changes`, we can do `save changes and keep as draft`, `save changes and announce event`, and `delete event`.

Or, on the list page, have `announce event` / `uncancel event` / `cancel` event (rather than have it on the edit page, which only has the `save changes` button)
<img width="838" alt="screen shot 2018-04-09 at 8 09 18 pm" src="https://user-images.githubusercontent.com/11889765/38529318-20450e84-3c32-11e8-93be-aa640cccdc6a.png">

ashleytqy commented on app/partials/actions/event/event-add.html:
```diff
@@ -18,7 +18,11 @@ <h5>{{rdesc.attributes.description}}</h5>
-		<button class="btn btn-primary" ng-click="createResource(model, rdesc)">Create</button>
+	<div class="form-group" ng-if="statusFound">
+		<button class="btn btn-primary" ng-click="createResource(model, rdesc, 'announced')">Announce Event</button>
+		<button class="btn btn-info" ng-click="createResource(model, rdesc, 'draft')">Draft</button>
```
Can we do `Save as Draft` instead of `Draft`?

ashley committed:
named submit buttons

ashley committed:
Fixed wording ine event-add button

ashley committed:
Change color of save and annouce event button

ashley commented:
- I renamed the buttons so that the original "Save Changes" button is now, "Save Changes as x event" where x is the current status.
- I also changed the announce event button to green so users can differentiate them more.

ashleytqy commented:
@ashley screenshots would be helpful here !

ashley commented:
**Event-Add Page**
<img width="1130" alt="screen shot 2018-04-10 at 7 30 59 pm" src="https://user-images.githubusercontent.com/2825071/38588816-c9c5c2f0-3cf5-11e8-9520-03521504f26f.png">

**Event-Edit Canceled Page**
<img width="1148" alt="screen shot 2018-04-10 at 7 30 06 pm" src="https://user-images.githubusercontent.com/2825071/38588813-c988637e-3cf5-11e8-929a-37bf3f5d9155.png">

**Event-Edit Announced Page**
<img width="1135" alt="screen shot 2018-04-10 at 7 30 25 pm" src="https://user-images.githubusercontent.com/2825071/38588814-c999a4cc-3cf5-11e8-98e4-0000980041b3.png">

**Event-Edit Draft Page**
<img width="1129" alt="screen shot 2018-04-10 at 7 30 41 pm" src="https://user-images.githubusercontent.com/2825071/38588815-c9ac9cbc-3cf5-11e8-88a5-0890257035c3.png">

ashleytqy reviewed (approved):
💯
